# 2. Análisis Exploratorio Inicial (EDA)

Trás definir el objetivo de la práctica procedemos a hacer el analisis exploratorio de los datos. 

En primer luga cargamos todas las librerias que necesitaremos y hacemos una visualización general de dataframe, de su composición y del tipo de datos que contiene cada columna.

In [1]:
import pandas as pd 
from pandas_profiling import ProfileReport
data = pd.read_csv('Melbourne_housing_FULL.csv')
%run -i fundamentos_datos_variables.py

In [2]:
data.head(20)

Suburb              Address  Rooms Type      Price Method  SellerG  \
0   Abbotsford        68 Studley St      2    h        NaN     SS   Jellis   
1   Abbotsford         85 Turner St      2    h  1480000.0      S   Biggin   
2   Abbotsford      25 Bloomburg St      2    h  1035000.0      S   Biggin   
3   Abbotsford   18/659 Victoria St      3    u        NaN     VB   Rounds   
4   Abbotsford         5 Charles St      3    h  1465000.0     SP   Biggin   
5   Abbotsford     40 Federation La      3    h   850000.0     PI   Biggin   
6   Abbotsford          55a Park St      4    h  1600000.0     VB   Nelson   
7   Abbotsford         16 Maugie St      4    h        NaN     SN   Nelson   
8   Abbotsford         53 Turner St      2    h        NaN      S   Biggin   
9   Abbotsford         99 Turner St      2    h        NaN      S  Collins   
10  Abbotsford       129 Charles St      2    h   941000.0      S   Jellis   
11  Abbotsford         124 Yarra St      3    h  1876000.0      S   Nelson   
12  Abbotsford  121/56 Nicholson St      2    u        NaN     PI   Biggin   
13  Abbotsford        17 Raphael St      4    h        NaN      W   Biggin   
14  Abbotsford        98 Charles St      2    h  1636000.0      S   Nelson   
15  Abbotsford     217 Langridge St      3    h  1000000.0      S   Jellis   
16  Abbotsford      18a Mollison St      2    t   745000.0      S   Jellis   
17  Abbotsford   6/241 Nicholson St      1    u   300000.0      S   Biggin   
18  Abbotsford        10 Valiant St      2    h  1097000.0      S   Biggin   
19  Abbotsford  403/609 Victoria St      2    u   542000.0      S   Dingle   

         Date  Distance  Postcode  ...  Bathroom  Car  Landsize  BuildingArea  \
0   3/09/2016       2.5    3067.0  ...       1.0  1.0     126.0           NaN   
1   3/12/2016       2.5    3067.0  ...       1.0  1.0     202.0           NaN   
2   4/02/2016       2.5    3067.0  ...       1.0  0.0     156.0          79.0   
3   4/02/2016       2.5    3067.0  ...       2.0  1.0       0.0           NaN   
4   4/03/2017       2.5    3067.0  ...       2.0  0.0     134.0         150.0   
5   4/03/2017       2.5    3067.0  ...       2.0  1.0      94.0           NaN   
6   4/06/2016       2.5    3067.0  ...       1.0  2.0     120.0         142.0   
7   6/08/2016       2.5    3067.0  ...       2.0  2.0     400.0         220.0   
8   6/08/2016       2.5    3067.0  ...       1.0  2.0     201.0           NaN   
9   6/08/2016       2.5    3067.0  ...       2.0  1.0     202.0           NaN   
10  7/05/2016       2.5    3067.0  ...       1.0  0.0     181.0           NaN   
11  7/05/2016       2.5    3067.0  ...       2.0  0.0     245.0         210.0   
12  7/11/2016       2.5    3067.0  ...       2.0  1.0    4292.0          82.0   
13  7/11/2016       2.5    3067.0  ...       2.0  0.0     230.0         147.0   
14  8/10/2016       2.5    3067.0  ...       1.0  2.0     256.0         107.0   
15  8/10/2016       2.5    3067.0  ...       NaN  NaN       NaN           NaN   
16  8/10/2016       2.5    3067.0  ...       NaN  NaN       NaN           NaN   
17  8/10/2016       2.5    3067.0  ...       1.0  1.0       0.0           NaN   
18  8/10/2016       2.5    3067.0  ...       1.0  2.0     220.0          75.0   
19  8/10/2016       2.5    3067.0  ...       NaN  NaN       NaN           NaN   

    YearBuilt         CouncilArea Lattitude  Longtitude  \
0         NaN  Yarra City Council  -37.8014    144.9958   
1         NaN  Yarra City Council  -37.7996    144.9984   
2      1900.0  Yarra City Council  -37.8079    144.9934   
3         NaN  Yarra City Council  -37.8114    145.0116   
4      1900.0  Yarra City Council  -37.8093    144.9944   
5         NaN  Yarra City Council  -37.7969    144.9969   
6      2014.0  Yarra City Council  -37.8072    144.9941   
7      2006.0  Yarra City Council  -37.7965    144.9965   
8      1900.0  Yarra City Council  -37.7995    144.9974   
9      1900.0  Yarra City Council  -37.7996    144.9989   
10        NaN  Yar

In [3]:
print("Cantidad de Filas y de Columnas",data.shape)


Cantidad de Filas y de Columnas (34857, 21)


In [4]:
data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 34857 entries, 0 to 34856
Data columns (total 21 columns):
 #   Column         Non-Null Count  Dtype  
---  ------         --------------  -----  
 0   Suburb         34857 non-null  object 
 1   Address        34857 non-null  object 
 2   Rooms          34857 non-null  int64  
 3   Type           34857 non-null  object 
 4   Price          27247 non-null  float64
 5   Method         34857 non-null  object 
 6   SellerG        34857 non-null  object 
 7   Date           34857 non-null  object 
 8   Distance       34856 non-null  float64
 9   Postcode       34856 non-null  float64
 10  Bedroom2       26640 non-null  float64
 11  Bathroom       26631 non-null  float64
 12  Car            26129 non-null  float64
 13  Landsize       23047 non-null  float64
 14  BuildingArea   13742 non-null  float64
 15  YearBuilt      15551 non-null  float64
 16  CouncilArea    34854 non-null  object 
 17  Lattitude      26881 non-null  float64
 18  Longti

Durante estas primeras observaciones ya se puede identificar que el dataset  es bastante grande con 34857 y 21 vcolumnas, pero también contiene datos faltantes en bastantes variables incluyendo la varibable objetivo "Price". Además, cuenta a priori con varibles cualitativas y cuantitivas.

Por lo tanto, para hacer un análisis en profundidad usamos la función "profiling" de pandas. 


In [5]:
profile = ProfileReport(data, title="Pandas Profiling Report")

In [6]:
profile

De esta función obtenemos informacion muy relevante con respecto al dataset:

+ El dataset cuenta con 13 variables numéricas y 8 categóricas 

+ Existe un pequeño porcentaje de datos duplicados, tomaremos esto en cuenta durante el análisis de variables para poder identificarlos y tratarlos. 

+ El pocetaje de datos faltantes corresponde a un 13,8%  y todo ellos se encuentran en las variables numéricas. (es alto es bajo). Siendo "YeatBuild" , "BuildingArea", y "Landsize".

+ Las variables con mayor correlación son: "rooms" y "Bedrrom2". Tambien podemos  observar que tanto ambas variables tienen una alta correlacion entre ellas y podrían producir colinealidad. 

Para poder clasificar los tipo de varibles vamos a realizar una función que las identifique que tipo de varaibles es cada una en función del tipo de datos que contiene. 

In [7]:

clasificar_variables(data)
   

h    23980
u     7297
t     3580
Name: Type, dtype: int64
¿Es ordinal?(seleccionar si o no): no
S     19744
SP     5095
PI     4850
VB     3108
SN     1317
PN      308
SA      226
W       173
SS       36
Name: Method, dtype: int64
¿Es ordinal?(seleccionar si o no): no
Southern Metropolitan         11836
Northern Metropolitan          9557
Western Metropolitan           6799
Eastern Metropolitan           4377
South-Eastern Metropolitan     1739
Eastern Victoria                228
Northern Victoria               203
Western Victoria                115
Name: Regionname, dtype: int64
¿Es ordinal?(seleccionar si o no): no


{'Suburb': 'Cualitativa Politoma',
 'Address': 'Cualitativa Politoma',
 'Rooms': 'Cuantitativa Discreta de Proporcion Politomizada No Acotada',
 'Type': 'Cualitativa Politoma No ordinal',
 'Price': 'Cuantitativa Continua de Proporcion Politomizada No Acotada',
 'Method': 'Cualitativa Politoma No ordinal',
 'SellerG': 'Cualitativa Politoma',
 'Date': 'Cualitativa Politoma',
 'Distance': 'Cuantitativa Continua de Proporcion Politomizada No Acotada',
 'Postcode': 'Cuantitativa Continua de Proporcion Politomizada No Acotada',
 'Bedroom2': 'Cuantitativa Continua de Proporcion Politomizada No Acotada',
 'Bathroom': 'Cuantitativa Continua de Proporcion Politomizada No Acotada',
 'Car': 'Cuantitativa Continua de Proporcion Politomizada No Acotada',
 'Landsize': 'Cuantitativa Continua de Proporcion Politomizada No Acotada',
 'BuildingArea': 'Cuantitativa Continua de Proporcion Politomizada No Acotada',
 'YearBuilt': 'Cuantitativa Continua de Proporcion Politomizada No Acotada',
 'CouncilArea': 

Esto nos será de mucha utilidad en el análisis de las que realizaremos a continuación y su posterior transformación. 